src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/ProjectPropertiesExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    internal static class ProjectPropertiesExtensions
+    {
+        /// <summary>
+        /// Saves the unevaluated value of <paramref name="propertyName"/>, if any, to <paramref name="storage"/>.
+        /// </summary>
+        public static async Task RememberValueIfCurrentlySet(this IProjectProperties properties, string propertyName, ITemporaryPropertyStorage storage)
+        {
+            string? currentPropertyValue = await properties.GetUnevaluatedPropertyValueAsync(propertyName);
+            if (!Strings.IsNullOrEmpty(currentPropertyValue))
```

Comment:
>Strings [](start = 17, length = 7)

why Strings here and not in the other method?

Reply:
Just an oversight. I'll make it consistent.